src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -183,35 +193,32 @@ QueryPlanPtr MergeTreeDataSelectExecutor::read(
             num_streams,
             max_block_numbers_to_read,
             query_info.projection->merge_tree_projection_select_result_ptr);
+    }
 
-        if (plan->isInitialized())
+    if (projection_plan->isInitialized())
+    {
+        if (query_info.projection->before_where)
         {
-            // If `before_where` is not empty, transform input blocks by adding needed columns
```

Comment:
lost comment?

Reply:
It's intentional. The comment is outdated. We use ActionsDAG now.